src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -36,8 +33,26 @@ private static string[] ParseMultiValue(string value)
                 }
                 else if ((value[i] == ',') && !inquote)
                 {
-                    singleValue = new string(vp, 0, chIndex);
-                    tempStringCollection.Add(singleValue.Trim());
+                    int start, end;
+
+                    for (start = 0; start < chIndex; ++start)
+                    {
+                        if (!char.IsWhiteSpace(vp[start]))
+                        {
+                            break;
+                        }
+                    }
+
+                    for (end = chIndex - 1; end >= start; ++end)
```

Comment:
Also, I think it'd be nice to move this all into a helper function like:
```C#
private static string SubstringTrim(char[] chars, int start, int length);
```
both so that it's reusable if necessary and so that it's more clear at the call site what's happening.